Improve documentation for history-graph #19893

koskila opened the pull request:
## Proposed change

#19893 shows documentation for history-graph might be lacking. This change makes it more clear how to customize how history-graph (and many other cards) figure out how to display different entities.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: #19893 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

koskila committed:
Improve documentation for history-graph #19893

Adds another sample and a link to instructions on how to customize unit_of_measurement for history-graph's shown entities.

frenck commented on source/_lovelace/history-graph.markdown:
```diff
@@ -59,7 +59,9 @@ name:
 {% endconfiguration %}
 
-## Example
+Changing the units used to generate the graphs requires changing the entities' unit_of_measurement property, as it can't be overridden on the history-graph card. For more details, see [Customizing Entities](/docs/configuration/customizing-devices/#unit_of_measurement).
```
This is, unfortunately, no longer true. State classes are also considered nowadays.
Additionally, we really don't want to recommend customizing devices, as we consider it bad practice.

koskila replied:
Thanks for your comments! Good to be learning more, thought I was up-to-speed here but I clearly wasn't!

I can see why customizing entities would be seen as bad practice - what's the best alternative, if one needs to change the unit of measurement for an entity they haven't explicitly created themselves?

frenck replied:
In those cases, I would say it is a bug and should be reported so it can be fixed (instead of worked around).

koskila replied:
Fair enough - thanks!

frenck commented on source/_lovelace/history-graph.markdown:
````diff
@@ -71,3 +73,15 @@ entities:
     name: Main player
 ```
+
+```yaml
````
Add a small description before the example to explain what it does / why it is here.

koskila replied:
Done 👍

frenck commented on source/_lovelace/history-graph.markdown:
````diff
@@ -71,3 +73,15 @@ entities:
+
+```yaml
+type: history-graph
+title: 'Temperatures in the last 48 hours'
````
```suggestion
title: "Temperatures in the last 48 hours"
```

frenck commented on source/_lovelace/history-graph.markdown:
```diff
@@ -71,3 +73,15 @@ entities:
+entities:
+  - sensor.outside_temperature
+  - entity: sensor.lounge_temperature
+    name: Lounge
```
```suggestion
    name: "Lounge"
```

frenck commented on source/_lovelace/history-graph.markdown:
```diff
@@ -71,3 +73,15 @@ entities:
+  - entity: sensor.lounge_temperature
+    name: Lounge
+  - entity: sensor.attic_temperature
+    name: Attic
```
```suggestion
    name: "Attic"
```

koskila committed:
Update source/_lovelace/history-graph.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

koskila committed:
Update source/_lovelace/history-graph.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

koskila committed:
Update source/_lovelace/history-graph.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

koskila committed:
Update according to suggestions

Remove the whole paragraph about unit of measurement - better not fiddle with it in regards to particular cards.

Add explanation for the second sample - I'm hoping to clarify the logic when the entities will be grouped and when not, as it is quite nifty.

frenck reviewed (approved):
Thanks, @koskila 👍 

Happy Hacktoberfest 🎉 

../Frenck
